### PR TITLE
feat: Twitter Login

### PR DESCRIPTION
**Pull request by berry-13**
# Added the possibility to login via Twitter.
### ⚠️I HAVE ADDED A FOLDER IN THE DOCUMENTS TO BETTER ORGANIZE THE FILES⚠️
Bugs:

- ~~Twitter icon is not visible.~~
- ~~It does not link the account to the existing one.~~
## Change Type

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

## Testing

test if the twitter login works correctly

### **Test Configuration**:

I've had some issues because Twitter does **NOT** allow the use of localhost, so I tested on a private domain to see if it worked.

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,7 +54,9 @@
   },
   "homepage": "https://github.com/danny-avila/LibreChat#readme",
   "dependencies": {
-    "axios": "^1.4.0"
+    "axios": "^1.4.0",
+    "nodemon": "3.0.1",
+    "passport-twitter": "^0.1.5"
```
Comment (danny-avila) [inline package.json]: nodemon and passport-twitter need to be uninstalled from the root directory and instead installed in the api directory

**Review by danny-avila — CHANGES_REQUESTED** [review]

**Comment by Berry-13** [comment]
I've read some documentation both on Twitter and on Twitter's Passport.js, and it seems that other people are also facing the same issue with receiving the email... I believe you need to request the app to be whitelisted and have a paid API, which I don't think anyone wants to do😅. Here are a few links to better understand.

[GET account/verify_credentials](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/manage-account-settings/api-reference/get-account-verify_credentials), [Github Issue 1](https://github.com/jaredhanson/passport-twitter/issues/67), [Github Issue 2](https://github.com/jaredhanson/passport-twitter/issues/97), [stack overflow](https://stackoverflow.com/questions/75180779/node-js-passport-twitter-unable-to-get-user-email-in-passport-twitter-authentic) and [Twitter Forum](https://twittercommunity.com/tags/c/oauth/12/email)

**Comment by Berry-13** [comment]
I will close this PR since no one is responding to the question I asked on the Twitter Developer Community. Therefore, it is impossible to proceed

**Comment by Berry-13** [comment]
I will close this PR since no one is responding to the question I asked on the Twitter Developer Community. Therefore, it is impossible to proceed

**Comment by danny-avila** [comment]
> I will close this PR since no one is responding to the question I asked on the Twitter Developer Community. Therefore, it is impossible to proceed

sorry to hear :(

**Pull request closed without merging**